File: src/main/java/com/bullhornatsclone/service/ClientService.java
package com.bullhornatsclone.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.bullhornatsclone.domain.Client;
import com.bullhornatsclone.dto.ClientDTO;
import com.bullhornatsclone.dto.ClientSearchDTO;
import com.bullhornatsclone.dto.ClientPageDTO;
import com.bullhornatsclone.dto.ClientConvertCriteriaDTO;
import com.bullhornatsclone.service.GenericService;
import com.bullhornatsclone.dto.common.RequestDTO;
import com.bullhornatsclone.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ClientService extends GenericService<Client, Integer> {

	List<Client> findAll();

	ResultDTO addClient(ClientDTO clientDTO, RequestDTO requestDTO);

	ResultDTO updateClient(ClientDTO clientDTO, RequestDTO requestDTO);

    Page<Client> getAllClients(Pageable pageable);

    Page<Client> getAllClients(Specification<Client> spec, Pageable pageable);

	ResponseEntity<ClientPageDTO> getClients(ClientSearchDTO clientSearchDTO);
	
	List<ClientDTO> convertClientsToClientDTOs(List<Client> clients, ClientConvertCriteriaDTO convertCriteria);

	ClientDTO getClientDTOById(Integer clientId);







}

File: bullhornatsclone_ui/src/services/ClientService.js
import http from "../http-common"; 

class ClientService {
  getAllClients(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/client/clients`, searchDTO);
  }

  get(clientId) {
    return this.getRequest(`/client/${clientId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/client?field=${matchData}`, null);
  }

  addClient(data) {
    return http.post("/client/addClient", data);
  }

  update(data) {
  	return http.post("/client/updateClient", data);
  }
  
  uploadImage(data,clientId) {
  	return http.postForm("/client/uploadImage/"+clientId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ClientService();
